runtime/sema/checker.go

```diff
@@ -1366,32 +1366,13 @@ func (checker *Checker) recordResourceInvalidation(
 		)
 	}
 
```

Comment:
The refactor of this function is probably best viewed in split-mode. Also, I first refactored the function a bit to simplify it in 7abf054ce0c4cadfe092d9736e173e18eadfb87d and to allow customization of the check what is allowed to a function in 48247ab22059162d0d97db8c0ca4ce715c2e1a90.